File: Geeks/Toggle.py
t=int(input())
arr=list(map(int,input().split()))[:t]
# firstPoint=pos=maxi=count=0
firstPoint=pos=maxi=count=0
# print(firstPoint,pos,maxi,count)
n=0
while pos < len(arr):
    if arr[pos] == 1:
        count +=1
        maxi = max(maxi,count)
        pos +=1
        n+=1
    else:
        while ( pos < len(arr) and arr[pos] ==0 ):
            count+=1
            pos+=1
            maxi = max(maxi,count)
            n+=1
            
        firstPoint = pos
        while ( pos < len(arr) and arr[pos] ==1 ):
            count +=1
            pos +=1
            maxi = max(maxi,count)
            n=pos
            
        pos=firstPoint
        count=0

print(maxi)
             
            

File: Geeks/Toggle.java
package Geeks;

import java.util.Scanner;

public class Toggle {
    static int subSet(int n,int arr[]){
        int start = 0,curr = 0;
        int mx = 0;
        int count = 0;
        //using sliding window
        while(curr<arr.length){
            if(arr[curr]==1){
                count++;
                mx = Math.max(mx,count);
                curr++;
            }
            else{
            while(curr<arr.length && arr[curr]==0){
                count++;
                curr++;
                mx = Math.max(mx,count);
            }
            start = curr;
            while(curr<arr.length&&arr[curr]==1){
                count++;
                curr++;
               mx = Math.max(mx,count);
            }
            curr=start;
            count=0;
            }
    }
    return mx;
    }
    public static void main(String[] args) {
        Scanner sc =new Scanner(System.in);
        int n=sc.nextInt();
        int a[]=new int[n];
        for(int i=0;i<n;i++)
        a[i]=sc.nextInt();
        System.out.println(subSet(n,a));
        sc.close();
    }
}
